File: backEnd/models/crime_model.js
const req = require('express/lib/request')
const mongoose = require('mongoose')
const Schema = mongoose.Schema


const crime_schema = mongoose.Schema({
    title:{type:String},
    location: {type:String,required:true},
    time: {type:String,required :true},
    description:{type:String,},
    imageURL:{ type:String},
    report_to:{type:String ,default:"kilinto"},
    status:{type:String },
    reporter: {type: Schema.Types.ObjectId, default:req.user, ref: 'User'},
})




crime_schema.methods.currentuser = function (req) {
    return this.reporter >= req.user;
}
const Crime = mongoose.model("Crime",crime_schema)


module.exports = Crime

File: front-end/lib/infrastructure/local_sqlite/crime_model.dart
class Crime {
  final int title;
  final String location;
  final String time;
  final String description;
  final int imageURL;
  final int report_to;
  final int status;
  final int reporter;

  Crime({ required this.title, required this.location, required this.time, required this.description, required this.imageURL, required this.report_to, required this.status, required this.reporter, });

  Map<String, dynamic> toMap() {
    return {
      'title': title,
      'location': location,
      'time': time,
      'description': description,
      'imageURL': imageURL,
      'report_to':report_to,
      'status': status,
      'reporter': reporter,
    };
  }

  @override
  String toString() {
    return 'Crime{title: $title, location: $location, time: $time, description: $description,imageURL :$imageURL, report_to: $report_to , status: $status , reporter: $reporter}';
  }

}